sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.AI.FormRecognizer.Models
+{
+    /// <summary>
+    /// </summary>
+    public struct FieldValue
+    {
+#pragma warning disable CS0649 // Add readonly modifier
+        private FieldValue_internal _fieldValue;
+#pragma warning restore CS0649 // Add readonly modifier
+
+        /// <summary> Type of field value. </summary>
+        public FieldValueType Type { get; internal set; }
+
+        /// <summary>
+        /// Gets the value of the field as a <see cref="string"/>.
+        /// </summary>
+        /// <returns></returns>
+        public string AsString()
+        {
+            if (Type != FieldValueType.String)
+            {
+                throw new InvalidOperationException($"Cannot get field as String.  Field value's type is {Type}.");
+            }
+
+            return _fieldValue.ValueString;
+        }
+
+        /// <summary>
+        /// Gets the value of the field as an <see cref="int"/>.
+        /// </summary>
+        /// <returns></returns>
+        public int AsInt32()
+        {
+            if (Type != FieldValueType.Integer)
+            {
+                throw new InvalidOperationException($"Cannot get field as Integer.  Field value's type is {Type}.");
+            }
+
+            if (!_fieldValue.ValueInteger.HasValue)
+            {
+                throw new InvalidOperationException($"Field value is null.");
+            }
+
+            return _fieldValue.ValueInteger.Value;
+        }
+
+        /// <summary>
+        /// Gets the value of the field as an <see cref="float"/>.
+        /// </summary>
+        /// <returns></returns>
+        public float AsFloat()
+        {
+            if (Type != FieldValueType.Number)
+            {
+                throw new InvalidOperationException($"Cannot get field as Float.  Field value's type is {Type}.");
+            }
+
+            if (!_fieldValue.ValueNumber.HasValue)
+            {
+                throw new InvalidOperationException($"Field value is null.");
+            }
+
+            return _fieldValue.ValueNumber.Value;
+        }
+
+        /// <summary>
+        /// Gets the value of the field as an <see cref="DateTimeOffset"/>.
+        /// </summary>
+        /// <returns></returns>
+#pragma warning disable CA1822
+        public DateTime AsDate()
+#pragma warning restore CA1822
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets the value of the field as an <see cref="DateTimeOffset"/>.
```

Comment:
nit: it should be as an `TimeSpan` here, correct?

Reply:
you are right. updated and modified English in other docstrings of this class.